pkg/recommendation/minmaxcapacity.go

```diff
@@ -0,0 +1,203 @@
+		// nodeCapacityDeviceCountMap
+		for nodeName, blockDeviceList := range nodeBlockDeviceListMap {
+
+			// If capacity count map not found for a node then create it.
+			capacityCountMap, capacityCountMapFound := nodeCapacityDeviceCountMap[nodeName]
+			if !capacityCountMapFound {
+				capacityCountMap = make(map[int64]int64)
+			}
+
+			// Iterate over blockDeviceList and populate
+			for _, blockDevice := range blockDeviceList {
+
+				// find capacity for each block device nodeBlockDeviceListMap
+				capacity, capacityFound := blockDevice.Capacity.AsInt64()
+				if !capacityFound {
+					// TODO handle this case
+					continue
+				}
+
+				// In capacity device count map if entry not found for any capacity then
+				// make it 0. If found then increase by 1.
+				_, ok := capacityCountMap[capacity]
+				if !ok {
+					capacityCountMap[capacity] = 0
+				}
+				capacityCountMap[capacity]++
+			}
+
+			// update nodeCapacityDeviceCountMap for each node.
+			nodeCapacityDeviceCountMap[nodeName] = capacityCountMap
+		}
+
+		result := &MinMaxCapacity{
+			MinCapacity: resource.Quantity{},
+			MaxCapacity: resource.Quantity{},
+		}
+
+		// Iterate over nodeCapacityDeviceCountMap and get min and max.
+		for _, capacityDeviceCountMap := range nodeCapacityDeviceCountMap {
+			for capacity, deviceCount := range capacityDeviceCountMap {
+
+				// If device count is less than group device count
+				// then skip those block devices.
+				if deviceCount < r.RaidGroupConfig.GroupDeviceCount {
+					continue
+				}
+
+				// effectiveBlockDeviceCount is round up block device count.
+				// ie - one node has 5 block devices of 100GB and pool type
+				// is mirror then effective block device count will be 4.
+				effectiveBlockDeviceCount := deviceCount / r.RaidGroupConfig.GroupDeviceCount
+
+				// Get min capacity from block device capacity and count.
+				// If min capacity in response 0 or greater than new min
+				// then update it.
+				min, err := resource.ParseQuantity(fmt.Sprintf("%d",
+					(capacity * r.RaidGroupConfig.GetDataDeviceCount())))
+				if err != nil {
+					continue
+				}
+				if result.MinCapacity.IsZero() || result.MinCapacity.Cmp(min) > 0 {
+					result.MinCapacity = min
+				}
+
+				// Get max capacity from block device capacity and count.
+				// If max capacity in response 0 or less than new max then
+				// update it.
+				max, err := resource.ParseQuantity(fmt.Sprintf("%d",
+					(capacity * (effectiveBlockDeviceCount / r.RaidGroupConfig.GroupDeviceCount) *
+						r.RaidGroupConfig.GroupDeviceCount)))
+				if err != nil {
+					continue
+				}
+				if result.MaxCapacity.IsZero() || result.MaxCapacity.Cmp(max) < 0 {
+					result.MaxCapacity = max
+				}
+			}
+		}
+
+		// put min max result for all the types in a map.
+		resultMap[kind] = result
+	}
+
+	response := &MinMaxCapacity{
+		MinCapacity: resource.Quantity{},
+		MaxCapacity: resource.Quantity{},
+	}
+
+	// Iterate over min max capacity map for each type of
```

Comment:
Is this step needed? Or should we just send out the minmax for each type?

Reply:
We can send this map as per our discussion.